hal/tests/TESTS/pin_names/pinvalidate.py

```diff
@@ -14,29 +14,62 @@
 limitations under the License.
 """
 
-import os
-import re
 import argparse
-import pathlib
 import json
+import os
+import pathlib
+import re
+import sys
 from tabulate import tabulate
+from enum import Enum
+
+
+class ReturnCode(Enum):
+    """Return codes."""
+
+    SUCCESS = 0
+    ERROR = 1
+    INVALID_OPTIONS = 2
+
+
+class TestCaseError(Exception):
+    """An exception for test case failure."""
+
+
+class ArgumentParserWithDefaultHelp(argparse.ArgumentParser):
+    """Subclass that always shows the help message on invalid arguments."""
+
+    def error(self, message):
+        """Error handler."""
+        sys.stderr.write("error: {}\n".format(message))
+        self.print_help()
+        raise SystemExit(ReturnCode.INVALID_OPTIONS.value)
+
+
+def find_target(target_name=""):
+    """Find a target."""
+    mbed_os_root = os.path.abspath(
+        os.path.join(
+            pathlib.Path(__file__).parent.absolute(), "..", "..", "..", ".."
+        )
+    )
 
-def find_target(target_name = ''):
-    mbed_os_root = os.path.abspath(os.path.join(pathlib.Path(__file__).parent.absolute(), '..', '..', '..', '..'))
     targets = dict()
-    
-    try:
-        targets_json_file = open(os.path.join(mbed_os_root, 'targets', 'targets.json'))
-        target_data = json.load(targets_json_file)
-    except:
-        exit("Failed to open targets.json")
 
-    for root, dirs, files in os.walk(mbed_os_root):
+    targets_json_file = open(
+        os.path.join(mbed_os_root, "targets", "targets.json")
+    )
+
+    target_data = json.load(targets_json_file)
+
+    for root, _, files in os.walk(mbed_os_root):
         for f in files:
             if f == "PinNames.h":
```

Comment:
`pathlib` takes care of this a little easier, unless I am misunderstanding something:

```python
for f in mbed_os_root.rglob("PinNames.h"):
```
The added benefit here would be that each f the generator yields would already contain the parent's name value, so your regex would operate on just `f.name` for the file name, or `f.parent.name` for the folder that `PinNames.h` is in.

I am not going to lie, the regex confuses me a little, but that is only because I am not a regex fiend. It is possible that I am missing something significant here, but I still believe `pathlib` is more of your friend than all of the `os.path.join` calls and similar methods.